### PR TITLE
wxGUI: easier switching between locations (GRASS GIS 8)

### PR DESCRIPTION
**Pull request by lindakladivova**
When switching to a map layer in a different location this message will appear:

Map layer in different location_004

### CODE REVIEW

**Review by neteler — APPROVED** [review]
This is a great solution! Thanks so much @lindakladivova

File: gui/wxpython/datacatalog/tree.py [hunk 1]
```diff
@@ -981,17 +982,31 @@ def OnDoubleClick(self, node):
                     dlg = wx.MessageDialog(
                         parent=self,
                         message=_(
-                            "Map <{0}@{1}> is not in the current location"
-                            " and therefore cannot be displayed."
-                            "\n\n"
-                            "To display this map switch to mapset <{1}> first."
+                            "Map <{0}@{1}> is not in the current location. "
+                            "Do you want to switch to the location '{2}'? "
+                            "Your currently displayed layers will disappear."
```
Comment (wenzeslaus) [inline gui/wxpython/datacatalog/tree.py:987]: The reason I used the word "disappear" in the original issue is that it expresses well the confusion a user may have when the layers are unloaded, i.e., something unexpected happened. Furthermore, "disappear" in this context may sound as "underlying data will be deleted". I don't have another suggestion at this point, but Perhaps mentioning again the reason for it in connection with the change in wording can bring some understanding.

**Review by wenzeslaus — CHANGES_REQUESTED** [review]

File: gui/wxpython/datacatalog/tree.py [hunk 2]
```diff
@@ -1838,30 +1860,30 @@ def SwitchMapset(self, grassdb, location, mapset, show_confirmation=False):
         """
         Switch to location and mapset interactively.
         """
-        if can_switch_mapset_interactive(self, grassdb, location, mapset):
```
Comment (petrasovaa) [inline gui/wxpython/datacatalog/tree.py:1841]: What is the reason for moving this if out of the function?

Comment (lindakladivova) [reply gui/wxpython/datacatalog/tree.py:1841]: The reason is that we display a layer only if a switching is successful. I had another idea how to solve this issue - make a optional argument display_layer for SwitchMapset method. But finally I realized that SwitchMapset is something not much related to displaying layer.
Now code also seems to me a bit clearer because SwitchMapset really switch in all cases.
Making a display_layer argument would be much shorter in code but I think not much meaningful in terms of logic.
Or we could return True or False from SwitchMapset but I am not sure If this function should return something, It is more of procedure. What do you think @petrasovaa ?

Comment (petrasovaa) [reply gui/wxpython/datacatalog/tree.py:1841]: I see. I would be in favor of returning true/false since we need to use can_switch_mapset_interactive anyway every time we want to switch it, so it may be better have it in the SwitchMapset method.

File: gui/wxpython/datacatalog/tree.py (hunk 1 above)
Comment (petrasovaa) [reply gui/wxpython/datacatalog/tree.py:987]: Maybe something along the lines "All current Map Displays will be closed"?

Comment (lindakladivova) [reply gui/wxpython/datacatalog/tree.py]: And put it on the next line? Now it looks as follows: switch

File: gui/wxpython/datacatalog/tree.py [hunk 3]
```diff
@@ -981,17 +982,25 @@ def OnDoubleClick(self, node):
                     dlg = wx.MessageDialog(
                         parent=self,
                         message=_(
-                            "Map <{0}@{1}> is not in the current location"
-                            " and therefore cannot be displayed."
-                            "\n\n"
-                            "To display this map switch to mapset <{1}> first."
+                            "Map <{0}@{1}> is not in the current location. "
+                            "Do you want to switch to the location '{2}'? \n\n"
```
Comment (petrasovaa) [inline gui/wxpython/datacatalog/tree.py:986]: Just a small change for consistency, it should be <location>, not 'location'.

Comment (wenzeslaus) [inline gui/wxpython/datacatalog/tree.py:986]: `{0} {1} {2}` should be `{map_name} {map_mapset} {map_location}`?

**Review by wenzeslaus — CHANGES_REQUESTED** [review]

File: gui/wxpython/datacatalog/tree.py [hunk 4]
```diff
@@ -981,17 +982,25 @@ def OnDoubleClick(self, node):
...
+                            "All current Map Displays will be closed."
```
Comment (wenzeslaus) [inline gui/wxpython/datacatalog/tree.py:987]: This looks like "will be closed in any case". I think the new paragraph needs to be between "...the current location" and "All current Map...".

**Review by wenzeslaus — CHANGES_REQUESTED** [review]

File: gui/wxpython/datacatalog/tree.py (hunk 4 above)
Comment (lindakladivova) [reply gui/wxpython/datacatalog/tree.py:987]: Ok, then I would suggest this formulation:
easier_switch

Comment (veroandreo) [reply gui/wxpython/datacatalog/tree.py]: That message sounds a bit intimidating to me... Can we say something like: 

```
Map <bla> is not in the current location. To be able to display it you need to switch to <blabla> location. Note that if you switch there all current Map Displays will be closed. 

Do you want to switch anyway?
```

Too long? My intention is to "explain" in short that to display a map from a different location the user needs to switch there.

Comment (lindakladivova) [reply gui/wxpython/datacatalog/tree.py]: I like your idea more Vero. It is true that it is quite long but we need to fully explain it. @petrasovaa do you agree with that?

Comment (nilason) [reply gui/wxpython/datacatalog/tree.py]: > That message sounds a bit intimidating to me... Can we say something like:
> 
> ```
> Map <bla> is not in the current location. To be able to display it you need to switch to <blabla> location. 
> Note that if you switch there all current Map Displays will be closed. 
> 
> Do you want to switch anyway?
> ```
> 
> Too long? My intention is to "explain" in short that to display a map from a different location the user needs to switch there.

I think @veroandreo 's formulation is a bit more appealing. However, I think the button labels are too wordy. What about `Cancel` (as default) and `Switch`?

Comment (lindakladivova) [reply gui/wxpython/datacatalog/tree.py]: switch
